src/vs/platform/terminal/common/xterm/shellIntegrationAddon.ts

```diff
@@ -244,6 +247,10 @@ export class ShellIntegrationAddon extends Disposable implements IShellIntegrati
 			commandDetection = new CommandDetectionCapability(terminal, this._logService);
 			this.capabilities.add(TerminalCapability.CommandDetection, commandDetection);
 		}
+		if (!this._hasUpdatedTelemetry) {
+			this._telemetryService?.publicLog2<{ classification: 'SystemMetaData'; purpose: 'FeatureInsight' }>('terminal/shellIntegrationActivated');
+			this._hasUpdatedTelemetry = true;
+		}
```

Comment:
More accurately would be to wrap `_handleVSCodeSequence` as it's possible a shell integration event could fire and not be the command detection one. Example:

```ts
private _handleVSCodeSequence(data: string): boolean {
	const didHandle = this._doHandleVSCodeSequence(data);
	if (didHandle) {
		// public log
	}
	return didHandle;
}
```